packages/frontend/src/components/ExecutionList.tsx

```diff
@@ -23,35 +23,48 @@ export function ExecutionList(props: ExecutionListProps) {
     loading: true,
     data: null,
   });
+  const pollingPeriod = 5000; // 5s
+
   const [searchParams] = useSearchParams();
   const api = useClient();
 
   useEffect(() => {
-    api
-      ?.GET("/executions", {
-        params: {
-          query: {
-            limit,
-            recordId,
-            offset: (parseInt(searchParams.get("page") ?? "1", 10) - 1) * limit,
-          },
-        },
-      })
-      .then((response) => {
-        setExecutions({
-          loading: false,
-          data: response.data ?? {
-            limit: 0,
-            offset: 0,
-            total: 0,
-            records: [],
+    function getExecutions() {
```

Comment:
nit: can we move this function all the way out from the `ExecutionList` component? And pass the `api`, `limit`, `recordId` etc as parameters to this function? 

Because once we have a function `async getExecutions({ api, limit, recordId ...  })` that returns a list of executions, so we can use it like:

```typescript
setInterval(async () => {
  const executions = await getExecutions({ api, limit, recordId });
  setExecutions({ loading: false, data: executions.data });
}, pollingPeriod);
```

in the `useEffect` hook, which is imo easier to understand (less nested code).